lib/ipam/ipam.go

```diff
@@ -548,7 +549,44 @@ func (s *blockAssignState) findOrClaimBlock(ctx context.Context, minFreeIps int)
 	return nil, false, errors.New("failed to find or claim a block")
 }
 
-func (c ipamClient) autoAssign(ctx context.Context, num int, handleID *string, attrs map[string]string, requestedPools []net.IPNet, version int, host string, maxNumBlocks int, rsvdAttr *HostReservedAttr) ([]net.IPNet, error) {
+type IPAMAssignmentInfo struct {
+	NumRequested       int
+	NumAssigned        int
+	IPVersion          int
+	NumBlocksOwned     int
+	MaxNumBlocks       int
+	ExhaustedPools     []string
+	StrictAffinity     bool
+	NoFreeAffineBlocks bool
+	HostReservedAttr   *HostReservedAttr
+}
+
+func (i *IPAMAssignmentInfo) String() string {
+	if i == nil {
```

Comment:
You don't actually need this check, even though it's a pointer receiver.

Here's an example: https://play.golang.org/p/YkUMSPgpFZN

Reply:
TIL as well, thanks!